src/diffusers/pipelines/stable_diffusion_xl/pipeline_stable_diffusion_xl.py

```diff
@@ -383,8 +383,13 @@ def encode_prompt(
                 negative_prompt_embeds_list.append(negative_prompt_embeds)
 
             negative_prompt_embeds = torch.concat(negative_prompt_embeds_list, dim=-1)
-
-        prompt_embeds = prompt_embeds.to(dtype=self.text_encoder_2.dtype, device=device)
+        if self.text_encoder_2 is not None:
+            target_dtype = self.text_encoder_2.dtype
+        elif self.unet is not None:
+            target_dtype = self.unet.dtype
+        else:
+            raise ValueError("No text encoder or unet found.")
```

Comment:
```suggestion
        target_dtype = self.unet.dtype
```
Nice catch - can we here actually always use the dtype from the `self.unet`? 

Also it would be incredible if you could also update the following pipeline:
- https://github.com/huggingface/diffusers/blob/cd21b965d1d10266d399b887f77b7dfc429f5f04/src/diffusers/pipelines/stable_diffusion_xl/pipeline_stable_diffusion_xl_inpaint.py#L534
-